src/index.ts

```diff
@@ -188,18 +189,32 @@ async function catchupAndWatchChain(
         return undefined;
       }
 
-      const res = await fetch(`${config.ipfsGateway}/ipfs/${cid}`, {
-        timeout: 2000,
-        retry: { retries: 10, maxTimeout: 60 * 1000 },
-        // IPFS data is immutable, we can rely entirely on the cache when present
-        cache: "force-cache",
-        cachePath:
-          config.cacheDir !== null
-            ? path.join(config.cacheDir, "ipfs")
-            : undefined,
-      });
-
-      return (await res.json()) as T;
+      const url = `${config.ipfsGateway}/ipfs/${cid}`;
+
+      chainLogger.trace(`Fetching ${url}`);
+
+      try {
+        const res = await fetch(url, {
+          timeout: 2000,
+          onRetry(cause) {
+            chainLogger.debug(`Retrying IPFS request ${String(cause)}`);
+          },
+          retry: { retries: 3, minTimeout: 2000, maxTimeout: 60 * 10000 },
+          // IPFS data is immutable, we can rely entirely on the cache when present
+          cache: "force-cache",
+          cachePath:
+            config.cacheDir !== null
+              ? path.join(config.cacheDir, "ipfs")
+              : undefined,
+        });
+
+        return (await res.json()) as T;
+      } catch (err) {
+        chainLogger.warn({
+          msg: "failed to load IPFS file",
```

Comment:
Is the `url` shown in the error already? If not, it would be good to have it in the log message.

Reply:
@bard I didn't realize we caught errors from the handle event, so this catch and log might not be needed here.

This is how the Fetch error I've been seeing looks like serialized so there's enough information to see what's happening:

```
{
      "type": "FetchError",
      "message": "network timeout at: https://ipfs-grants-stack.gitcoin.co/ipfs/bafkreig7gzi3mlj5bcfbrzakisvchdnpok7blwvhs54dsjfcavbct6yrwi",
      "stack":
          FetchError: network timeout at: https://ipfs-grants-stack.gitcoin.co/ipfs/bafkreig7gzi3mlj5bcfbrzakisvchdnpok7blwvhs54dsjfcavbct6yrwi
              at Timeout.<anonymous> (/Users/moboudra/dev/gitcoin/allo-indexer/node_modules/minipass-fetch/lib/index.js:108:18)
              at listOnTimeout (node:internal/timers:564:17)
              at process.processTimers (node:internal/timers:507:7)
      "code": "FETCH_ERROR",
      "errno": "FETCH_ERROR"
}
```

Reply:
Yeah, unfortunately "just crash" was untenable in production when the cause of the crash was guaranteed to appear on the next start.